src/Grpc/HttpApi/src/Microsoft.AspNetCore.Grpc.HttpApi/Internal/Json/Int64Converter.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Type = System.Type;
+
+namespace Microsoft.AspNetCore.Grpc.HttpApi.Internal.Json;
+
+internal sealed class Int64Converter : JsonConverter<long>
+{
+    public override long Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        if (reader.TokenType == JsonTokenType.String)
+        {
+            return long.Parse(reader.GetString()!, CultureInfo.InvariantCulture);
```

Comment:
Is support for string numbers really needed?

Reply:
It's a rule in gRPC <-> JSON transcoding. 64 bits don't fit into a JavaScript's 50ish bit number. So send as a string instead.

Not a fan but need to support it.